src/firefly/js/ui/tap/TapUtil.js

```diff
@@ -188,9 +202,25 @@ async function doLoadTapTables(serviceUrl, schemaName) {
         }
         const schemaNameIdx = getColumnIdx(tableModel, 'schema_name');
         const tableNameIdx = getColumnIdx(tableModel, 'table_name');
+        const descIdx = getColumnIdx(tableModel, 'table_desc');
+        const doSwap= descIdx<tableNameIdx;
 
         const rTable = omit(tableModel, 'tableData.data');
         rTable.tableData.data = tableModel.tableData.data.filter((row) => row[schemaNameIdx] === schemaName);
+        if (doSwap) {
+            rTable.tableData.data = rTable.tableData.data
```

Comment:
I see why it's needed.  But, this workaround looks pretty bad. Maybe the fix is at `loadSchemaDefJoin`.  But, that's just a guess.

Reply:
If we could talk the table to reorder the columns then we could avoid it. I also considered doing it in the select but there are other reasons to do a `select *`.